### PR TITLE
game page cleanup

### PR DESCRIPTION
**Pull request by Leptopoda**
small cleanup of the gamepage code

The commits will be sqashed upon merge

### CODE REVIEW

File: src/ui/flutter_app/lib/screens/game_page/game_page.dart [hunk 1]
```diff
@@ -62,131 +59,93 @@ class _GamePageState extends State<GamePage>
...
-    debugPrint("$_tag Check if need to set Ready state...");
-    if (!isReady && mounted) {
-      debugPrint("$_tag Set Ready State...");
-      setState(() {});
-      isReady = true;
-      timer.cancel();
-
-      if (Localizations.localeOf(context).languageCode == "zh" &&
-          !LocalDatabaseService.preferences.isPrivacyPolicyAccepted) {
-        onShowPrivacyDialog();
```
Comment (calcitem) [inline src/ui/flutter_app/lib/screens/game_page/game_page.dart:89]: Why was the privacy statement dialog box removed? As I mentioned in other posts earlier, according to China’s information security laws, each app must display a privacy statement dialog box when it is first launched.

**Review by calcitem — CHANGES_REQUESTED** [review]
AI is very easy to hang or crash.

See: https://github.com/calcitem/Sanmill/issues/418

**Comment by Leptopoda** [comment]
There shouldn't be any logic change in here. 

Seems like the performance improvements just shows the underlaying race conditions...

File: src/ui/flutter_app/lib/screens/game_page/game_page.dart (hunk 1 above)
Comment (Leptopoda) [reply src/ui/flutter_app/lib/screens/game_page/game_page.dart:89]: and I mentioned earlier that I didn't remove it just cleaned the code ;)

I fixed it and restructured it :upside_down_face:

**Comment by Leptopoda** [comment]
ok I have quite a few other changes I'm currently working on (rewriting the mess of a code in game_page).
It is progressing fast and verry well but I'll close this one in favor of the other because even If we could fix this problem the two branches would be to much appart.

**Pull request closed without merging**